torch/profiler/_utils.py

```diff
@@ -45,9 +46,9 @@ def intervals_overlap(self, intervals: List[Interval]):
         overlap_time = 0
         intervals = sorted(intervals, key=lambda x: x.start)
         for i, interval in enumerate(intervals):
-            if i + 1 < len(intervals):
-                assert interval.end <= intervals[
-                    i + 1].start, "Intervals must be disjoint"
+            if i + 1 < len(intervals) and interval.end > intervals[i +
```

Comment:
Why was this changed from an assert to a condition?

Reply:
Because i actually came across two over lapping interval when I apply this in decrease interval. For the sake of convenience, I thought I could just change one of the interval's end to be non-overlapping.

Reply:
Ah yes, that was what I was doing here. But I have reverted this change, because that only occurs after I introduce the decrease interval calculation. I will think about it in a later commit.